src/components/Scrollbar.tsx

```diff
@@ -0,0 +1,158 @@
+/*
+Copyright 2018 - 2022 The Alephium Authors
+This file is part of the alephium project.
+
+The library is free software: you can redistribute it and/or modify
+it under the terms of the GNU Lesser General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+The library is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+GNU Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public License
+along with the library. If not, see <http://www.gnu.org/licenses/>.
+*/
+
+// Used as reference: https://github.com/xobotyi/react-scrollbars-custom/issues/46#issuecomment-897506147
+
+import React, { CSSProperties, useCallback, useRef, useState, WheelEvent } from 'react'
+import Scrollbar, { ScrollbarProps } from 'react-scrollbars-custom'
+import { ElementPropsWithElementRef, ScrollState } from 'react-scrollbars-custom/dist/types/types'
+import { useTheme } from 'styled-components'
+
+import { ScrollContextProvider } from '../contexts/scroll'
+
+const paddingRight = '6px'
+const width = `calc(6px + ${paddingRight})`
+
+const paddingTop = '6px'
+const height = `calc(6px + ${paddingTop})`
+
+const createScrollbarPiece = (regular: CSSProperties, additional?: CSSProperties, useAdditionalStyling?: boolean) => ({
+  renderer: ({ elementRef, style, ...restProps }: ElementPropsWithElementRef) => (
+    <div {...restProps} ref={elementRef} style={{ ...style, ...regular, ...(useAdditionalStyling && additional) }} />
+  )
+})
+
+interface ScrollbarCustomProps extends ScrollbarProps {
+  // When children have dynamic sizes, set this to true (like the AddressSummaryCards in OverviewPage/Header).
+  isDynamic?: boolean
+}
+
+const ScrollbarCustom = (props: ScrollbarCustomProps) => {
+  const theme = useTheme()
+  const [scroll, setScroll] = useState<ScrollState>()
+  const [isScrolling, setIsScrolling] = useState(false)
+  const [isMouseOver, setIsMouseOver] = useState(false)
+  const scrollerElemRef = useRef<HTMLDivElement | null>(null)
+
+  const isShow = isScrolling || isMouseOver
+
+  const onScrollStart = () => setIsScrolling(true)
+  const onScrollStop = () => setIsScrolling(false)
+  const onMouseEnter = () => setIsMouseOver(true)
+  const onMouseLeave = () => setIsMouseOver(false)
+
+  const { isDynamic, noScrollX } = props
```

Comment:
nit: Usually we deconstruct props next to the component definition, although it's not a hard rule, so all good here. Later on, however, I see that you are using props like `props.elementRef` or `props.noScrollY`, but here you're deconstructing only some of them (`isDynamic, noScrollX`). How about we either deconstruct all that we need here, or better, at the component definition? 😇

Reply:
I wouldn't deconstructed `elementRef` or `noScrollY` because it's just adding duplicate text since they're only used once to pass through, but absolutely this could move to being deconstructed in the function arguments